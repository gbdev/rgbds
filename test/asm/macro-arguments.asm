MACRO mac
	println "'mac \#':"
	for i, _NARG
		println strfmt("\\%d: <\1>", i+1)
		shift
	endr
	println
ENDM

	mac /* block
		...comment */ ; comment
	mac /*a*/ 1 , 2 /*b*/ , ; trailing comma
	mac \
	c, d
	mac 1, 2 + /* another ;
		; comment */ 2, 3

	mac           a                                                                                                                                                                                                                                                               b           ; truncated

	mac
	mac ,
	mac a,
	mac a,,
	mac ,,z
	mac a,,z
	mac ,a,b,c,
	mac ,,x,,,
	mac E,O,F ; no newline