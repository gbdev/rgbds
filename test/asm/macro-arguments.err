warning: macro-arguments.asm(21): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(23): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(24): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(24): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(25): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(26): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(27): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(27): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(27): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(27): [-Wempty-macro-arg]
    Empty macro argument
