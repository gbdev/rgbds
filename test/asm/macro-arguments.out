'mac ':

'mac  1,2':
\1: < 1>
\2: <2>

'mac c,d':
\1: <c>
\2: <d>

'mac 1,2 +  2,3':
\1: <1>
\2: <2 +  2>
\3: <3>

'mac a                                                                                                                                                                                                                                                               b':
\1: <a                                                                                                                                                                                                                                                               b>

'mac ':

'mac ,':
\1: <>

'mac a':
\1: <a>

'mac a,,':
\1: <a>
\2: <>

'mac ,,z':
\1: <>
\2: <>
\3: <z>

'mac a,,z':
\1: <a>
\2: <>
\3: <z>

'mac ,a,b,c':
\1: <>
\2: <a>
\3: <b>
\4: <c>

'mac ,,x,,,':
\1: <>
\2: <>
\3: <x>
\4: <>
\5: <>

'mac E,O,F':
\1: <E>
\2: <O>
\3: <F>

